parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -586,108 +595,105 @@ Type getType(PrimitiveTypeName type) {
+    switch (type.getSetField()) {
       case MAP:
-        return ConvertedType.MAP;
-      case MAP_KEY_VALUE:
-        return ConvertedType.MAP_KEY_VALUE;
-      case LIST:
-        return ConvertedType.LIST;
-      case ENUM:
-        return ConvertedType.ENUM;
-      case DECIMAL:
-        return ConvertedType.DECIMAL;
+        return LogicalTypeAnnotation.MapLogicalTypeAnnotation.create();
+      case BSON:
+        return LogicalTypeAnnotation.BsonLogicalTypeAnnotation.create();
       case DATE:
-        return ConvertedType.DATE;
-      case TIME_MILLIS:
-        return ConvertedType.TIME_MILLIS;
-      case TIME_MICROS:
-        return ConvertedType.TIME_MICROS;
-      case TIMESTAMP_MILLIS:
-        return ConvertedType.TIMESTAMP_MILLIS;
-      case TIMESTAMP_MICROS:
-        return ConvertedType.TIMESTAMP_MICROS;
-      case INTERVAL:
-        return ConvertedType.INTERVAL;
-      case INT_8:
-        return ConvertedType.INT_8;
-      case INT_16:
-        return ConvertedType.INT_16;
-      case INT_32:
-        return ConvertedType.INT_32;
-      case INT_64:
-        return ConvertedType.INT_64;
-      case UINT_8:
-        return ConvertedType.UINT_8;
-      case UINT_16:
-        return ConvertedType.UINT_16;
-      case UINT_32:
-        return ConvertedType.UINT_32;
-      case UINT_64:
-        return ConvertedType.UINT_64;
+        return LogicalTypeAnnotation.DateLogicalTypeAnnotation.create();
+      case ENUM:
+        return LogicalTypeAnnotation.EnumLogicalTypeAnnotation.create();
       case JSON:
-        return ConvertedType.JSON;
-      case BSON:
-        return ConvertedType.BSON;
+        return LogicalTypeAnnotation.JsonLogicalTypeAnnotation.create();
+      case LIST:
+        return LogicalTypeAnnotation.ListLogicalTypeAnnotation.create();
+      case TIME:
+        TimeType time = type.getTIME();
+        return LogicalTypeAnnotation.TimeLogicalTypeAnnotation.create(time.isAdjustedToUTC, convertTimeUnit(time.unit));
+      case STRING:
+        return LogicalTypeAnnotation.StringLogicalTypeAnnotation.create();
+      case DECIMAL:
+        DecimalType decimal = type.getDECIMAL();
+        return LogicalTypeAnnotation.DecimalLogicalTypeAnnotation.create(decimal.scale, decimal.precision);
+      case INTEGER:
+        IntType integer = type.getINTEGER();
+        return LogicalTypeAnnotation.IntLogicalTypeAnnotation.create(integer.bitWidth, integer.isSigned);
+      case UNKNOWN:
+        return null;
+      case TIMESTAMP:
+        TimestampType timestamp = type.getTIMESTAMP();
+        return LogicalTypeAnnotation.TimestampLogicalTypeAnnotation.create(timestamp.isAdjustedToUTC, convertTimeUnit(timestamp.unit));
       default:
-        throw new RuntimeException("Unknown original type " + type);
-     }
-   }
+        throw new RuntimeException("Unknown logical type " + type);
+    }
+  }
+
+  LogicalTypeAnnotation.TimeUnit convertTimeUnit(TimeUnit unit) {
```

Comment:
As far as I can see this method can be private. Please, use visibility as tight as possible and comment if something is visible for testing purposes only.